MSVC: Static gflags library doesn't work

```
Benchmark.h
DECLARE_bool(benchmark);

Benchmark.cpp
DEFINE_bool(benchmark, false, "Run benchmarks.");

Benchmark.cpp(35): error C2370: 'gflags::FLAGS_benchmark': redefinition; different storage class
Benchmark.h(34): note: see declaration of 'gflags::FLAGS_benchmark'
```
Because `DEFINE_VARIABLE` always add `GFLAGS_DLL_DEFINE_FLAG`

--- a/src/gflags.h.in
+++ b/src/gflags.h.in
@@ -86,7 +86,7 @@
 
 // We always want to export variables defined in user code
 #ifndef GFLAGS_DLL_DEFINE_FLAG
-#  ifdef _MSC_VER
+#  if GFLAGS_IS_A_DLL && defined(_MSC_VER)
 #    define GFLAGS_DLL_DEFINE_FLAG __declspec(dllexport)
 #  else
 #    define GFLAGS_DLL_DEFINE_FLAG

